src/auto/injector.js

```diff
@@ -843,15 +832,12 @@ function createInjector(modulesToLoad, strictDi) {
         fn = fn[fn.length - 1];
       }
 
-      if (!isClass(fn)) {
+      try {
         // http://jsperf.com/angularjs-invoke-apply-vs-switch
-        // #5388
         return fn.apply(self, args);
-      } else {
-        args.unshift(null);
-        /*jshint -W058 */ // Applying a constructor without immediate parentheses is the point here.
-        return new (Function.prototype.bind.apply(fn, args));
-        /*jshint +W058 */
+      } catch (e) {
```

Comment:
We could optimize the try-catch only for Safari 9 but I agree that there is a concern about running the constructor twice...